api/src/main/java/io/strimzi/api/kafka/model/listener/v2/GenericKafkaListenerConfiguration.java

```diff
@@ -0,0 +1,168 @@
+
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("Reference to the `Secret` which holds the certificate and private key pair which will be used for this listener. " +
+            "The certificate can optionally contain the whole chain. " +
+            "This field can be used only with listeners with enabled TLS encryption.")
+    public CertAndKeySecretSource getBrokerCertChainAndKey() {
+        return brokerCertChainAndKey;
+    }
+
+    public void setBrokerCertChainAndKey(CertAndKeySecretSource brokerCertChainAndKey) {
+        this.brokerCertChainAndKey = brokerCertChainAndKey;
+    }
+
+    @Description("Configures the `Ingress` class that defines which `Ingress` controller will be used. " +
+            "If not set, the `Ingress` class is set to `nginx`. " +
+            "This field can be used only with `ingress` type listener.")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonProperty("class")
+    public String getIngressClass() {
+        return ingressClass;
+    }
+
+    public void setIngressClass(String ingressClass) {
+        this.ingressClass = ingressClass;
+    }
+
+    @Description("Defines which address type should be used as the node address. " +
+            "Available types are: `ExternalDNS`, `ExternalIP`, `InternalDNS`, `InternalIP` and `Hostname`. " +
+            "By default, the addresses will be used in the following order (the first one found will be used):\n" +
+            "* `ExternalDNS`\n" +
+            "* `ExternalIP`\n" +
+            "* `InternalDNS`\n" +
+            "* `InternalIP`\n" +
+            "* `Hostname`\n" +
+            "\n" +
+            "This field can be used to select the address type which will be used as the preferred type and checked first. " +
+            "In case no address will be found for this address type, the other types will be used in the default order." +
+            "This field can be used only with `nodeport` type listener.")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public NodeAddressType getPreferredAddressType() {
+        return preferredAddressType;
+    }
+
+    public void setPreferredAddressType(NodeAddressType preferredAddressType) {
+        this.preferredAddressType = preferredAddressType;
+    }
+
+    @Description("Specifies whether the service routes external traffic to node-local or cluster-wide endpoints. " +
+            "`Cluster` may cause a second hop to another node and obscures the client source IP. " +
+            "`Local` avoids a second hop for LoadBalancer and Nodeport type services and preserves the client source IP (when supported by the infrastructure). " +
+            "If unspecified, Kubernetes will use `Cluster` as the default." +
+            "This field can be used only with `loadbalancer` type listener.")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public ExternalTrafficPolicy getExternalTrafficPolicy() {
```

Comment:
Again, I think `externalTrafficPolicy` is a Kubernetes field, so I think it is best to keep it that way. I also found out it could be used with node ports as well and updated the description.